hosted/alibaba/p0/p0_provisioning_test.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright © 2023 - 2024 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package p0_test
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/rancher/shepherd/clients/rancher"
+	management "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+
+	"github.com/rancher/hosted-providers-e2e/hosted/alibaba/helper"
+	"github.com/rancher/hosted-providers-e2e/hosted/helpers"
+)
+
+var _ = Describe("P0Provisioning", func() {
+	for _, testData := range []struct {
+		qaseID    int64
+		isUpgrade bool
+		testBody  func(cluster *management.Cluster, client *rancher.Client, clusterName string)
+		testTitle string
+	}{
+		{
+			qaseID:    318,
+			isUpgrade: false,
+			testBody:  p0NodesChecks,
+			testTitle: "should successfully Add, scale up/down the Alibaba hosted cluster nodes & Delete it",
+		},
+		{
+			qaseID:    321,
+			isUpgrade: true,
+			testBody:  p0upgradeK8sVersionChecks,
+			testTitle: "should be able to upgrade k8s version of the provisioned Alibaba hosted cluster and Delete it",
+		},
+	} {
+		testData := testData
+		When("an Alibaba hosted cluster is created", func() {
+			BeforeEach(func() {
+				if testData.isUpgrade && helpers.SkipUpgradeTests {
```

Comment:
I think we need to update the `helpers.SkipUpgradeTests`. The current definition will always return false.
```go
	SkipUpgradeTests = func() bool {
		// Skip upgrade tests since lowest k8s version not available
		return strings.Contains((RancherFullVersion), "2.8")
	}()
    SkipTest = func() bool {
		// Some features not available on v2.8, v2.9
		return strings.Contains((RancherFullVersion), "2.8") || strings.Contains((RancherFullVersion), "2.9")
	}()
```

Same goes with `SkipTest`.

Reply:
I've removed the conditional upgrade, since it is not required for alibaba.

This is also not required in other hosted cluster, so will remove for those as well in other PR.